core/templates/pages/admin-page/navbar/admin-navbar.directive.spec.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for admin navbar component.
+ */
+
+import { importAllAngularServices } from 'tests/unit-test-utils';
+
+describe('Admin Navbar component', () => {
+  var $rootScope = null;
+  var ctrl = null;
+  var $scope = null;
+  var $q = null;
+  var UserService = null;
+  var AdminRouterService = null;
+  var userProfileImage = 'profile-data-url';
+  var userInfo = {
+    isModerator: () => true,
+    getUsername: () => 'username1',
+    isSuperAdmin: () => true
+  };
+  var imagePath = '/path/to/image.png';
+  var profileUrl = '/profile/username1';
+
+  importAllAngularServices();
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    UserService = $injector.get('UserService');
+    AdminRouterService = $injector.get('AdminRouterService');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $q = $injector.get('$q');
+    ctrl = $componentController('adminNavbar', {
+      $scope: $scope
+    });
+  }));
+
+  beforeEach(() => {
+    spyOn(UserService, 'getProfileImageDataUrlAsync')
+      .and.returnValue($q.resolve(userProfileImage));
+    spyOn(UserService, 'getUserInfoAsync').and.returnValue(
+      $q.resolve(userInfo));
+    ctrl.$onInit();
+    $rootScope.$apply();
+  });
+
+  it('should initialize ctrl properties after directive is initialized', () => {
```

Comment:
```suggestion
  it('should initialize component properties correctly', () => {
```

Can we convert this directive into a component?